### PR TITLE
feat(cli): API requests will respect HTTP_PROXY env variable

### LINKED ISSUE #574: Dealing with a lhci server behind proxy (create new projet)
I've just set up a `lhci-server` following the [documentation](https://github.com/GoogleChrome/lighthouse-ci/blob/main/docs/recipes/docker-server/README.md) and I am trying to create a new project using `lhci wizard` locally. 

The server is both reachable in my browser and a `curl` (after setting up my proxy: `HTTPS_PROXY` env variable) but I've got a `TIMEOUT` when trying to create a new project. 

```shell
➜  curl https://my-lhci-server/version
0.7.0%       
```

But when I go through the `lhci wizard`: 
```shell 
➜  lhci wizard
? Which wizard do you want to run? new-project
? What is the URL of your LHCI server? https://my-lhci-server
? What would you like to name the project? my-project
? Where is the project's code hosted? https://github.com/whatever
? What branch is considered the repo's trunk or main branch? master
FetchError: request to https://my-lhci-server/v1/projects failed, reason: connect ETIMEDOUT 35.204.214.248:443
    at ClientRequest.<anonymous> (/usr/local/lib/node_modules/@lhci/cli/node_modules/node-fetch/lib/index.js:1461:11)
    at ClientRequest.emit (node:events:376:20)
    at TLSSocket.socketErrorListener (node:_http_client:486:9)
    at TLSSocket.emit (node:events:376:20)
    at emitErrorNT (node:internal/streams/destroy:188:8)
    at emitErrorCloseNT (node:internal/streams/destroy:153:3)
    at processTicksAndRejections (node:internal/process/task_queues:80:21)%       
```

If I try to `curl` over the projects HTTP endpoint, it works: 

```shell
➜  curl https://my-lhci-server/v1/projects
[]%  
```

I guess it is related to the proxy configuration which isn't used by `lhci` when creating the project (but also when running) so I may be blocked because of it. 
Is there any way to provide a proxy url to `lhci` so it go through it to do any request? 

Thank you in advance for your help!

### PR DESCRIPTION
**Pull request by nicksrandall**
fixes #574

### CODE REVIEW

File: packages/cli/src/fetch.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
```
Comment (connorjclark) [inline packages/cli/src/fetch.js:2]:
```suggestion
 * @license Copyright 2021 Google Inc. All Rights Reserved.
```

**Comment by connorjclark** [comment]
Thanks! @nicksrandall can you sign the CLA please?

**Comment by connorjclark** [comment]
@nicksrandall CLA?

**Comment by nicksrandall** [comment]
Screenshot 2022-09-13 at 11 54 34 AM

**Comment by connorjclark** [comment]
ahhhh, just had to re-run the actions 😅

**Pull request merged**